### DISCUSSION TITLE
Best way to list all the dependencies of a repository

### DISCUSSION THREAD

**Discussion by Bkstandukar** [community/community#24330] [Apps, API and Webhooks]
<p>What could be the best way to list all the dependencies by just entering a repo name? I was trying to do this via GHTorrent (via GoogleBigQuery) but I assume it only shows events and not the actual dependencies of the project. Graphql may be the way to go. But can anyone lighten me up how can i achieve this? Isn’t there limitation to number of request for APIs? Also do I need OAuth token to access such manifest data from repositories? If that is the case, then graphql would not help my data analysis at all, as i aim to list dependencies for any given project in github.</p>

**Comment by canuckjacq** [comment, marked as answer]
<p>:wave: Welcome!</p>
<p>It looks like it’s possible <a href="https://docs.github.com/en/graphql/reference/objects#dependencygraphmanifestconnection">with our GraphQL API</a>.</p>
<p>It looks as though some clever person has put together a nice little tool for just this purpose. I haven’t tested it, but you can view the code yourself before installing:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/edsu/xkcd2347" target="_blank" rel="noopener">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:400/400;"><img src="https://avatars2.githubusercontent.com/u/33829?s=400&amp;amp;v=4" class="thumbnail" width="400" height="400"></div>

<h3><a href="https://github.com/edsu/xkcd2347" target="_blank" rel="noopener">edsu/xkcd2347</a></h3>


  <p><span class="label1">Get dependencies for a project on GitHub. Contribute to edsu/xkcd2347 development by creating an account on GitHub.</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>In terms of rate limiting, with Graphql queries, there’s some math involved. Here’s the explainer:</p>
<aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/graphql/overview/resource-limitations" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/graphql/overview/resource-limitations" target="_blank" rel="noopener">Resource limitations - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/graphql/overview/resource-limitations" target="_blank" rel="noopener">//docs.github.com/en/graphql/overview/resource-limitations</a></p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
